server/src/main/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldMapper.java

```diff
@@ -1052,6 +1070,16 @@ public String toString() {
         boolean supportsElementType(ElementType elementType) {
             return elementType != ElementType.BYTE;
         }
+
+        @Override
+        boolean updatableTo(IndexOptions update) {
+            boolean sameType = update.type.equals(this.type);
+            if (sameType) {
+                Int8HnswIndexOptions int8HnswIndexOptions = (Int8HnswIndexOptions) update;
+                sameType = int8HnswIndexOptions.m >= this.m;
```

Comment:
Could you add a comment as to why we need to verify `m`?

The reasoning is because when merging segments, we utilize the largest previous graph, which may have `m` connections. Then requiring fewer connections on later writes would be a bad idea as our assumptions about how many maximum connections there are would break.